include/zephyr/drivers/i2s.h

```diff
@@ -142,6 +142,50 @@ typedef uint8_t i2s_fmt_t;
  */
 #define I2S_FMT_DATA_FORMAT_RIGHT_JUSTIFIED (4 << I2S_FMT_DATA_FORMAT_SHIFT)
 
+/**
+ * @brief Time Division Multiplexed (TDM) mode A Data Format.
+ *
+ * Serial data is transmitted with the MSB first. Both Frame Synchronization
+ * (WS) and Serial Data (SD) signals are sampled on the rising edge of the
+ * clock (SCK) signal. The bits within the data word are left justified.
+ * The frame sync is one clock cycle long.
```

Comment:
I'm not really familiar with the TDM mode and I haven't used it myself yet, but from what I've read in several chip specifications, it seems to me that TDM should not be a separate format, but rather an option that could be applied to formats like I2S or PCM/DSP. In particular, I couldn't find any standard that would define that for TDM the above signal specification must be used (@blemouzy what document is the above specification based on?). Instead, I found for example NXP chips (LPC55S0x/LPC550x), where TDM can be combined with both the I2S and DSP modes. So maybe adding TDM as an `I2S_OPT_*` option would be more suitable?

Reply:
@anangl although in my mind this is not intuitive your answer is correct. 

I posted the same question to linux alsa-devel and got a similar answer as yours: 

https://www.spinics.net/lists/alsa-devel/msg165660.html

Reply:
Yeah, you are both right.

I think the correct names should be:
* `I2S_FMT_DATA_FORMAT_TDM_MODE_A` -->`I2S_FMT_DATA_FORMAT_DSP_A` = `I2S_FMT_DATA_FORMAT_PCM_SHORT`
* `I2S_FMT_DATA_FORMAT_TDM_MODE_B` -->`I2S_FMT_DATA_FORMAT_DSP_B`